File: shared/src/main/java/ru/privatenull/pnauth/config/DialogSettings.java
package ru.privatenull.pnauth.config;

public record DialogSettings(
        boolean enabled,
        boolean fallbackToCommands,
        boolean allowPlayerPreference,
        int minClientProtocol
) {
    public DialogSettings {
        if (minClientProtocol < 0) {
            throw new IllegalArgumentException("minClientProtocol must not be negative");
        }
    }

    public static DialogSettings defaults() {
        return new DialogSettings(true, true, true, 771);
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/config/DialogSettings.kt
package ru.privatenull.pnauth.config

@JvmRecord
data class DialogSettings(
    val enabled: Boolean,
    val fallbackToCommands: Boolean,
    val allowPlayerPreference: Boolean,
    val minClientProtocol: Int
) {
    init {
        require(minClientProtocol >= 0) { "minClientProtocol must not be negative" }
    }

    companion object {
        @JvmStatic
        fun defaults(): DialogSettings {
            return DialogSettings(true, true, true, 771)
        }
    }
}
